File: javascript/ch5/EitherFactory.js
class EitherFactory {
  static left(value) {
    const { Left } = require('./Left');
    return new Left(value);
  }

  static right(value) {
    const { Right } = require('./Right');
    return new Right(value);
  }

  static fromNullable(value) {
    return value != null ? EitherFactory.right(value) : EitherFactory.left(value);
  }

  static of(value) {
    return EitherFactory.right(value);
  }
}

module.exports = {
  EitherFactory
};

File: python/python/ch5/EitherFactory.py
# from .Left import Left
# from .Right import Right

class EitherFactory:
    @staticmethod
    def left(value):
        from .Left import Left
        return Left(value)
      
    @staticmethod
    def right(value):
        from .Right import Right
        return Right(value)
      
    @staticmethod
    def from_nullable(value):
        return EitherFactory.right(value) if value != None else EitherFactory.left(value)
      
    @staticmethod
    def of(value):
        return EitherFactory.right(value)